src/reducer.js

```diff
@@ -79,10 +80,20 @@ const behaviors = {
   [INITIALIZE](state, {data, fields, overwriteValues}) {
     return createInitialState(data, fields, state, overwriteValues);
   },
-  [REMOVE_ARRAY_VALUE](state, {path, index}) {
+  [REMOVE_ARRAY_VALUE](state, {path, indexOrPredicate}) {
```

Comment:
suggestion: it would be better if the param 'indexOrPredicate' would split into 2 different params, the first 1 is 'index' and the second is 'predicate', and the logic is basically the same.

when you check isFunction the indexOrPredicate, you'll check if (predicate && isFuncion(predicate))

the reason for the split is to keep Single Responsibility, each param should have only 1 purpose, and the word 'or' applies that this param have 2 different purposes

it's just a suggestion :)